### ISSUE TITLE
Support release branches

### ISSUE THREAD

**Issue by AZielinsky95** [steamclock/coda-changelog#11]
We should be adding release/* branches to our workflow files
and using the branch name to map to a table name.

**Comment by jenncoop** [comment]
Some projects that rely on release branches are not able to use the tool atm, so seems worth moving this one up a bit.

**Comment by AZielinsky95** [comment]
Those projects can use the tool!

All that needs to be done is the workflow file needs to be specified to run on any release branch. 

```
on:
  push:
    branches: [ release/* ]
```

**Comment by jenncoop** [comment]
Okay, wonderful! If there are multiple release branches at once, will all commits from each branch go into the single table?

**Comment by AZielinsky95** [comment]
Yes they will. The way around that currently, is to have multiple workflow files for each release branch that passes in a specific table name.

Do we ever write to more than one release branch at a time? 

One of the things we can do if we think it's worth it, is to parse the last path component of the release branch and look for a corresponding table.  So a  "release/1.5" branch would look for a table called "Release 1.5" if it couldn't find a table it would default to writing to Unreleased Changes table. (current default if no table name is provided)

**Comment by jenncoop** [comment]
@ssawchenko Do you ever have multiple release branches in flight at once?

**Comment by ssawchenko** [comment]
> @ssawchenko Do you ever have multiple release branches in flight at once?

I don't think we have this too often in practice, but in theory I could see a case where we have a release in the works but we need to apply a hotfix/patch to the current release.

ie. 
* Current release is 5.1
* We are doing active dev on `release/5.2`
* Bug comes in and we need to create `release/5.1.1`

In this case we would have 2 release branches in flight at once, unless we treat patches/hotfixes differently. I'm not sure we've hashed out a naming scheme for patches yet, so it's possible simply adopting a `patch/<number>` naming scheme would solve this.

**Comment by jenncoop** [comment]
@AZielinsky95 IIRC, the Coda API doesn't allow us to create tables, so we'd need to manually add the table with the corresponding release name ahead of the release starting. Am I remembering that correctly?

**Comment by AZielinsky95** [comment]
Yeah that's correct
